cmd/influxd/run/server.go

```diff
@@ -383,6 +382,9 @@ func (s *Server) Close() error {
 	if s.HintedHandoff != nil {
 		s.HintedHandoff.Close()
 	}
+	if s.Monitor != nil {
```

Comment:
I went that route first, but it became awkward due to circular dependencies. I didn't try super hard, but I think it's because, like `meta` it both uses the TSDB *and* provides query services to it. I am open to seeing if I can make it work though.

That said it seems most intuitive to me that it should be a service, but I am not sure. I may revisit this later.